lib/membrane/rtp/session_bin.ex

```diff
@@ -609,17 +609,14 @@ defmodule Membrane.RTP.SessionBin do
     state = %{state | ssrcs: Map.delete(state.ssrcs, ssrc)}
 
     to_remove =
-      [
-        :rtx_funnel,
-        :rtx,
-        :rtx_decryptor,
-        :stream_receive_bin
-      ]
-      |> Enum.map(&{&1, ssrc})
-      |> Enum.filter(fn name ->
-        child = Map.get(ctx.children, name)
-        child && not child.terminating?
+      ctx.children
+      |> Enum.filter(fn {name, child} ->
+        # we use pattern matching to match all
+        # extensions spawned for given ssrc stream
+        # e.g. vad
+        match?({_name, ^ssrc}, name) && not child.terminating?
       end)
+      |> Enum.map(fn {name, _child} -> name end)
```

Comment:
nitpick: I'd probably use the pattern in function head and maybe `flat_map` to save one iteration (assuming `terminating?` cannot be `nil`)
```suggestion
      ctx.children
      |> Enum.flat_map(fn 
        {{name, ^ssrc}, %{terminating?: false}} -> [name]
        _other -> []
      end)
```